polar-core/src/polar.rs

```diff
@@ -36,6 +37,19 @@ impl Query {
     pub fn next_message(&self) -> Option<Message> {
         self.vm.messages.next()
     }
+
+    pub fn source(&self) -> String {
+        let mut source_str = self.vm.term_source(&self.source);
+        if let Some(source) = self.vm.source(&self.source) {
```

Comment:
Could you explain what's going on here? It's a bit hard to follow - the `Query#source` method gets the `term_source` for its own `source` field which calls `VM#source` to get the term source info and then we call `VM#source` again (???), passing in the query's `source` field again... and I've already lost the plot of what this is actually supposed to result in 😄

Reply:
Renamed a bunch of stuff.